lib/mozart_fetcher/http_client.ex

```diff
@@ -39,6 +39,40 @@ defmodule HTTPClient do
     client.get(endpoint, headers, options)
   end
 
+  defp handle_response({:ok, response}) do
+    {:ok,
+     %HTTPoison.Response{
+       headers: response.headers,
+       body:
+         decode_response_body(response.body, content_encoding(process_headers(response.headers))),
+       status_code: response.status_code
+     }}
+  end
+
+  defp handle_response(response), do: response
+
+  defp process_headers(headers) do
+    Enum.map(headers, fn {k, v} -> {String.downcase(k), v} end)
+  end
+
+  defp decode_response_body(body, "gzip"), do: :zlib.gunzip(body)
+  defp decode_response_body(body, "x-gzip"), do: :zlib.gunzip(body)
```

Comment:
not a big deal at all, I just expected these functions to do different things. Maybe it'd read better having a guard to check the content type?

Reply:
I could do. I'll leave this for now as this will be refactored into separate modules in the near future.